Sources/AirRobeWidget/Views/OptInView/AirRobeOptInViewModel.swift

```diff
@@ -58,11 +59,19 @@ final class AirRobeOptInViewModel {
             )
             AirRobeUtils.dispatchEvent(eventName: EventName.pageView.rawValue, pageName: PageName.product.rawValue)
             alreadyInitialized = true
-            if category.isEmpty {
-                isAllSet = .paramIssue
-                return
+        }
+
+        if category.isEmpty {
+            isAllSet = .paramIssue
+            return
+        }
+        let eligibility = AirRobeShoppingDataModelInstance.shared.categoryMapping.checkCategoryEligible(items: [category])
+        if isAllSet == .eligible {
+            if !(eligibility.eligible && !shoppingDataModel.isBelowPriceThreshold(department: department, price: priceCents)) {
+                isAllSet = .notEligible
+                AirRobeUtils.dispatchEvent(eventName: EventName.widgetNotRendered.rawValue, pageName: PageName.product.rawValue)
             }
-            let eligibility = AirRobeShoppingDataModelInstance.shared.categoryMapping.checkCategoryEligible(items: [category])
+        } else {
```

Comment:
Does the category eligibility logic belong in a view model? It seems like a general data concern that isn't view specific?

Reply:
I see that this is calling out to a couple of data model methods around price + category eligibility. The fact that this work is being done twice in this file makes me think it could be abstracted?

Reply:
Could you add more details?